services/datamanager/builtin/builtin.go

```diff
@@ -431,8 +437,28 @@ func (svc *builtIn) Update(ctx context.Context, cfg *config.Config) error {
 		svc.closeCollectors()
 		return err
 	}
-	if cfg.Cloud != nil {
-		svc.partID = cfg.Cloud.ID
+
+	// Check that we have models to download and appropriate credentials.
+	if svcConfig.ModelsToDeploy != nil && cfg.Cloud != nil {
```

Comment:
Want to instead check if len(svcConfig.ModelsToDeploy) > 0 rather than nil, since if this gets initialized in Update and `models_to_deploy` is set to [], this will not be nil but there still won't be any models to deploy.